version: "3.1"
services:
  auth-redis:
    image: redis/redis-stack-server:latest
    networks:
      - main

  auth-database:
    image: mysql:latest
    environment:
      - MYSQL_ROOT_PASSWORD=nxhung
    restart: on-failure
    volumes:
      - "./scripts/db:/docker-entrypoint-initdb.d"
    networks:
      - main
    healthcheck:
      test: ["CMD", "mysqladmin" ,"ping", "-h", "localhost"]
      timeout: 10s
      retries: 10

  auth-service:
    build:
      context: ./
    environment:
      - MYSQL_HOST=auth-database
      - MYSQL_PASSWORD=nxhung
      - EUREKA_HOST=http://eureka:8761/eureka
      - JWT_SIGNING_KEY=01f63c787b72997c1622d3b1c74f7d857d3f1df1f2926cc5da8e4a6dbb10779e
      - REDIS_HOST_URL=auth-redis
      - REDIS_HOST_PORT=6379
    depends_on:
      auth-database:
        condition: service_healthy
      auth-redis:
        condition: service_started
    networks:
      - main

networks:
  main:
    driver: bridge

volumes:
  mysql:









